mobx/lib/src/core/reaction_helper.dart

```diff
@@ -146,24 +146,49 @@ ReactionDisposer createReaction<T>(ReactiveContext context,
 
 ReactionDisposer createWhenReaction(ReactiveContext context,
     bool Function(Reaction) predicate, void Function() effect,
-    {String name, void Function(Object, Reaction) onError}) {
+    {String name, int timeout, void Function(Object, Reaction) onError}) {
```

Comment:
with the `timeout` option, when will throw error if the predicate does not execute within the given time